pkg/utils/tekton/pipelineruns.go

```diff
@@ -0,0 +1,480 @@
+		logs, _ := utils.GetContainerLogs(ki, d.PodName, d.FailedContainerName, pipelineRun.Namespace)
+		failMessage += fmt.Sprintf("Logs from failed container '%s': \n%s", d.FailedContainerName, logs)
+	}
+	return failMessage, nil
+}
+
+// StoreFailedPipelineRunLogs stores logs and parsed yamls of pipelineRuns into directory of pipelineruns' namespace under ARTIFACT_DIR env.
+// In case the files can't be stored in ARTIFACT_DIR, they will be recorder in GinkgoWriter.
+func StoreFailedPipelineRun(pipelineRun *v1beta1.PipelineRun, c crclient.Client, ki kubernetes.Interface) error {
+	wd, _ := os.Getwd()
+	artifactDir := utils.GetEnv("ARTIFACT_DIR", fmt.Sprintf("%s/tmp", wd))
+	testLogsDir := fmt.Sprintf("%s/%s", artifactDir, pipelineRun.GetNamespace())
+
+	pipelineRunLog, err := GetFailedPipelineRunLogs(c, ki, pipelineRun)
+	if err != nil {
+		return fmt.Errorf("failed to store PipelineRun: %+v", err)
+	}
+
+	pipelineRunYaml, prYamlErr := yaml.Marshal(pipelineRun)
+	if prYamlErr != nil {
+		g.GinkgoWriter.Printf("\nfailed to get pipelineRunYaml: %s\n", prYamlErr.Error())
+	}
+
+	err = os.MkdirAll(testLogsDir, os.ModePerm)
+
+	if err != nil {
+		g.GinkgoWriter.Printf("\n%s\nFailed pipelineRunLog:\n%s\n---END OF THE LOG---\n", pipelineRun.GetName(), pipelineRunLog)
+		if prYamlErr == nil {
+			g.GinkgoWriter.Printf("Failed pipelineRunYaml:\n%s\n", pipelineRunYaml)
+		}
+		return err
+	}
+
+	filename := fmt.Sprintf("%s-pr-%s.log", pipelineRun.Namespace, pipelineRun.Name)
+	filePath := fmt.Sprintf("%s/%s", testLogsDir, filename)
+	if err := os.WriteFile(filePath, []byte(pipelineRunLog), 0644); err != nil {
+		g.GinkgoWriter.Printf("cannot write to %s: %+v\n", filename, err)
+		g.GinkgoWriter.Printf("\n%s\nFailed pipelineRunLog:\n%s\n", filename, pipelineRunLog)
+	}
+
+	if prYamlErr == nil {
+		filename = fmt.Sprintf("%s-pr-%s.yaml", pipelineRun.Namespace, pipelineRun.Name)
+		filePath = fmt.Sprintf("%s/%s", testLogsDir, filename)
+		if err := os.WriteFile(filePath, pipelineRunYaml, 0644); err != nil {
+			g.GinkgoWriter.Printf("cannot write to %s: %+v\n", filename, err)
+			g.GinkgoWriter.Printf("\n%s\nFailed pipelineRunYaml:\n%s\n", filename, pipelineRunYaml)
+		}
+	}
+
+	return nil
+}
+
+// StorePipelineRuns stores logs of all pipelineruns in given namespace in folder specified under ARTIFACT_DIR env.
+func (t *TektonController) StorePipelineRuns(componentPipelineRun *v1beta1.PipelineRun, testLogsDir, testNamespace string) error {
+	if err := os.MkdirAll(testLogsDir, os.ModePerm); err != nil {
+		return err
+	}
+
+	filepath := fmt.Sprintf("%s/%s-pr-%s.log", testLogsDir, testNamespace, componentPipelineRun.Name)
+	pipelineLogs, err := t.GetPipelineRunLogs(componentPipelineRun.Name, testNamespace)
+	if err != nil {
+		g.GinkgoWriter.Printf("got error fetching PR logs: %v\n", err.Error())
+	} else {
+		if err := os.WriteFile(filepath, []byte(pipelineLogs), 0644); err != nil {
+			g.GinkgoWriter.Printf("cannot write to %s: %+v\n", filepath, err)
+		}
+	}
```

Comment:
do we need this code?
this looks like a duplication of what also happens for this pipelinerun in a [for loop below](https://github.com/redhat-appstudio/e2e-tests/pull/688/files#diff-dda2163aa8e3e9cb6cd6f38a5e3df0a184e263ff34e8b7ee17ebe635d83325e1R345)

Reply:
Agree, will remove it.

Reply:
Removed.